java/code/src/com/suse/manager/webui/services/impl/SaltService.java

```diff
@@ -226,6 +239,72 @@ public void setReactor(SaltReactor reactorIn) {
         }
     }
 
+    @Override
+    public Optional<Map<String, JsonElement>> getCapabilities(String minionId) {
+        LocalCall<Map<String, JsonElement>> call =
+                new LocalCall<>("virt.all_capabilities", Optional.empty(), Optional.empty(),
+                        new TypeToken<Map<String, JsonElement>>() { });
+
+        return callSync(call, minionId);
+    }
+
+    @Override
+    public Map<String, JsonObject> getNetworks(String minionId) {
+        Map<String, Object> args = new LinkedHashMap<>();
+        LocalCall<Map<String, JsonElement>> call =
+                new LocalCall<>("virt.network_info", Optional.empty(), Optional.of(args),
+                        new TypeToken<Map<String, JsonElement>>() { });
+
+        Optional<Map<String, JsonElement>> nets = callSync(call, minionId);
+        Map<String, JsonElement> result = nets.orElse(new HashMap<String, JsonElement>());
+
+        // Workaround: Filter out the entries that don't match since we may get a retcode=0 one.
+        return result.entrySet().stream()
+                .filter(entry -> entry.getValue().isJsonObject())
+                .collect(Collectors.toMap(entry -> entry.getKey(), entry -> entry.getValue().getAsJsonObject()));
+    }
+
+    /**
+     * {@inheritDoc}
```

Comment:
Nitpick: I'd suggest to either have this tag on all public methods or just remove it everywhere.